pyproject.toml

```diff
@@ -1,119 +1,106 @@
-    "black",   # Syntax checking and reformatting
-    "pylint",  # Syntax checking
-    "flake8",  # Syntax checking
-    "flake8-rst-docstrings",  # numpy docstring checking
-    "flake8-docstrings",  # docstring checking
-    "flake8-rst",  # RST checking
-    "prettier",  # YAML, JSON, etc linting
-    "doc8",  # RST linting
-]
-debug = [
-    "ipython",
-]
-
-[project.entry-points."geoips.plugin_packages"]
-geoips = "geoips"
+keywords = ["GeoIPS", "Poetry"]
+packages = [{include = "geoips"}]  # not needed as tool.poetry.name is properly set already
+classifiers = [                # // list of PyPI trove classifiers to describe the project
+    "Programming Language :: Python :: 3", ]
 
-[project.entry-points.console_scripts]
-run_procflow = "geoips.commandline.run_procflow:main"
-convert_trackfile_to_yaml = "geoips.commandline.convert_trackfile_to_yaml:main"
-update_tc_tracks_database = "geoips.commandline.update_tc_tracks_database:main"
-xml_to_yaml_sector = "geoips.commandline.xml_to_yaml_sector:main"
-test_interfaces = "geoips.commandline.test_interfaces:main"
-list_available_modules = "geoips.commandline.list_available_modules:main"
-create_sector_image = "geoips.commandline.create_sector_image:main"
+# uses current version .postN and not incrementing
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+metadata = false    # don't include local-version hash; date doesn't appear by deafult
+#style = "pep440"    # pep440 is deafult; can rmeove later on; used for : keeping 0.0.8 as the default install until 0.0.9 is out
+style = "semver"
+#tag-branch = "feature-poetry"
+strict = true
+pattern = "(?x)^((?P<epoch>\\d+)!)?(?P<base>\\d+(\\.\\d+)*)([-._]?((?P<stage>[a-zA-Z]+)[-._]?(?P<revision>\\d+)?))?(\\+(?P<tagged_metadata>.+))?$"
+                         #           also: 3.6.0 incompatible with 0.20.3, but 0.21.0 works
+
+
+[tool.poetry-dynamic-versioning.substitution]
+files = ["*/_version.py"]
+
+
+[build-system]
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning>=1.0.0,<2.0.0"]
+build-backend = "poetry_dynamic_versioning.backend"     # object used to perform the build process
 
-[project.entry-points."geoips.readers"]
+[tool.poetry.dependencies]     # must download to run
+    python = ">=3.9, <3.11"         # mandatory to declare the required python version
+    matplotlib = "^3.7.2"           # Base requirement works, version specific to test outputs
+    netcdf4 = "^1.6.4"              # Base requirement
+    numpy = "^1.25.2"               # Base requirement
+    pyresample = "^1.27.1"          # Base requirement Geospatial image resampling 
+    				     	# efficiency improvements >= 1.22.3
+    pyyaml = "^6.0.1"               # Base requirement 
+    scipy = "^1.11.2"  	     # Base requirement, requires openblas 
+    xarray = "^2023.7.0"            # Base requirement
+    
+    astropy = "^5.3.2"              # Used for accounting leap seconds when converting epoch timestamps
```

Comment:
@jsolbrig I think I would prefer allowing versions to increment, and not tying to the major version number.  I like to see when things break, and then resolve dependency issues as they break, rather than getting way out of date and never noticing.  Thoughts?

Reply:
Yeah, I agree that we shouldn't be pinning everything. We should go back and remove all of the pinned versions that aren't already in the current pyproject.toml.

Reply:
@mindyls @jsolbrig the commented requirement is added in the latest commit on the feature-poetry [branch](https://github.com/SRIKANTH-KUMAR/geoips/commit/ec9650b26382be8e852fd242a5ebdb8fc656eb9b)

Reply:
This is fixed. Thanks @SRIKANTH-KUMAR